lib/storekit/error.rb

```diff
@@ -25,5 +25,41 @@ def initialize(status_code)
     def server_error?
       [21004, 21007, 21008].include?(status_code) || message == 'Unknown error'
     end
+
+    def self.new status_code
+      return super if self != ValidationError
+
+      @status_code = status_code
+      type =
+      case status_code
+      when 21000
+        InvalidJsonError
+      when 21002
+        InvalidReceiptFormatError
+      when 21003
+        ReceiptAuthenticationError
+      when 21004
+        SharedSecretInvalidError
+      when 21005
+        ReceiptServerUnavailableError
+      when 21006
+        SubscriptionExpiredError
+      when 21007
+        TestEnvironmentRequiredError
+      when 21008
+        ProductionEnvironmentRequiredError
+      end
+
+      type.is_a?(NilClass) && super || type.new(status_code)
+    end
   end
+
+  class InvalidJsonError < ValidationError; end
+  class InvalidReceiptFormatError < ValidationError; end
+  class ReceiptAuthenticationError < ValidationError; end
+  class SharedSecretInvalidError < ValidationError; end
```

Comment:
Since the first two new error classes have the `Invalid` prefix, let's do the same for this one as well.